src/wayland/tablet_manager/tablet.rs

```diff
@@ -0,0 +1,98 @@
+use std::ops::Deref as _;
+use std::path::PathBuf;
+use std::{cell::RefCell, rc::Rc};
+
+use wayland_protocols::unstable::tablet::v2::server::zwp_tablet_seat_v2::ZwpTabletSeatV2;
+use wayland_protocols::unstable::tablet::v2::server::zwp_tablet_v2::ZwpTabletV2;
+use wayland_server::protocol::wl_surface::WlSurface;
+use wayland_server::Filter;
+
+use crate::backend::input::Device;
+
+/// Description of graphics tablet device
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+pub struct TabletDescriptor {
+    /// Tablet device name
+    pub name: String,
+    /// Tablet device USB product id
+    pub id_product: Option<u32>,
+    /// Tablet device USB vendor id
+    pub id_vendor: Option<u32>,
+    /// Path to the device
+    pub syspath: Option<PathBuf>,
+}
+
+impl<D: Device> From<D> for TabletDescriptor {
+    fn from(device: D) -> Self {
+        TabletDescriptor {
+            name: device.name(),
+            syspath: device.syspath(),
+            id_product: device.id_product(),
+            id_vendor: device.id_vendor(),
+        }
+    }
+}
+
+#[derive(Debug, Default)]
+struct Tablet {
+    instances: Vec<ZwpTabletV2>,
+}
+
+/// Handle to a tablet device
+///
+/// Tablet represents one graphics tablet device
+#[derive(Debug, Default, Clone)]
+pub struct TabletHandle {
+    inner: Rc<RefCell<Tablet>>,
+}
+
+impl TabletHandle {
+    pub(super) fn new_instance(&mut self, seat: &ZwpTabletSeatV2, tablet: &TabletDescriptor) {
+        if let Some(client) = seat.as_ref().client() {
+            let wl_tablet = client
+                .create_resource::<ZwpTabletV2>(seat.as_ref().version())
+                .unwrap();
+
+            wl_tablet.quick_assign(|_, _req, _| {});
+
+            let inner = self.inner.clone();
+            wl_tablet.assign_destructor(Filter::new(move |instance: ZwpTabletV2, _, _| {
+                inner
+                    .borrow_mut()
+                    .instances
+                    .retain(|i| !i.as_ref().equals(&instance.as_ref()));
+            }));
+
+            seat.tablet_added(&wl_tablet);
+
+            wl_tablet.name(tablet.name.clone());
+
+            if let (Some(id_product), Some(id_vendor)) = (tablet.id_product, tablet.id_vendor) {
+                wl_tablet.id(id_product, id_vendor);
+            }
```

Comment:
What should happen if only one of `id_product` and `id_vendor` is `Some()`? Is this something that can happen?

Reply:
I'm not sure to be hones, when it comes to `libinput` both `id_product` and `id_vendor` are always returned (maybe those default to `0`, but there is no mention of that in the doc).

Should we perhaps send `0` if one of ID's was not supplied?

Reply:
If we always get both, I think we could store that as an `Option((u32, u32))`, rather than a pair of options, no?

Reply:
Good point, I'll do that.